src/libraries/Microsoft.Extensions.Caching.Memory/src/CacheEntry.cs

```diff
@@ -235,27 +237,40 @@ internal void SetExpired(EvictionReason reason)
             DetachTokens();
         }
 
-        private bool CheckForExpiredTime(DateTimeOffset now)
+        private bool CheckForExpiredTime(in DateTimeOffset now)
         {
-            if (_absoluteExpiration.HasValue && _absoluteExpiration.Value <= now)
-            {
-                SetExpired(EvictionReason.Expired);
-                return true;
-            }
+            if (_absoluteExpiration == null && _absoluteExpiration == null)
+                return false;
 
-            if (_slidingExpiration.HasValue
-                && (now - LastAccessed) >= _slidingExpiration)
+            return SlowPath(now);
```

Comment:
Is the idea here that the above check can get inlined, and then the “SlowPath” code remains a normal method call? Is there a good guidance on when this pattern should be used?

Side nit - is “SlowPath” a good name for this method?

Reply:
> Is the idea here that the above check can get inlined, and then the “SlowPath” code remains a normal method call?

Yes, exactly.

> Is there a good guidance on when this pattern should be used?

I would say that it should be used when the condtion typically returns false and rarely executes the code inside it (like throwing exceptions) and is ofc on a hot path (small helper methods executed frequently)